chat/uikit/android.md

```diff
@@ -0,0 +1,246 @@
+|`delivering`|Delivering text|String, default "Delivering"|
+|`delivered`|Delivered text|String, default "Delivered"|
+|`someRead`|Some read text|String, default "Some Read"|
+|`allRead`|All read text|String, default "All Read"|
+|`failed`|Failed text|String, default "Failed"|
+|`hint`|Place holder of input text|String, default "Type a message..."|
+|`statusTextColorForIncomingMessages`|Status text color (Incoming)|Color|
+|`statusTextColorForOutgoingMessages`|Status text color (Outgoing)|Color|
+|`timeTextColorForOutgoingMessages`|Time text color (Incoming)|Color|
+|`timeTextColorForIncomingMessages`|Time text color (Outgoing)|Color|
+
+
+##### Feature On/Off
+|Attribute|Description|Type|
+|---------|-----------|------|
+|`cameraButtonShouldShow`|Show camera button if true|Boolean, default: True|
+|`voiceMessageButtonShouldShow`|Show voice message button if true|Boolean, default: True|
+|`messageStatusShouldShow`|Show message status if true|Boolean, default: True|
+
+
+#### Event Hooks
+UIKit provides event hooks which enable you to implement specific feature when an event is fired.
+
+The below code segment passes a listener to a `ConversationActivity`.
+
+```kotlin
+i.putExtra(ConversationActivity.MessageFetchListenerIntentKey, object: MessageFetchListener {
+    override fun onBeforeMessageFetch(fragment: ConversationFragment) {
+
+    }
+
+    override fun onMessageFetchFailed(fragment: ConversationFragment, error: Error) {
+        Toast.makeText(fragment.activity, "Message Loading Failed", Toast.LENGTH_LONG).show()
+    }
+
+    override fun onMessageFetchSuccess(fragment: ConversationFragment, messages: List<Message>, isCached: Boolean) {
+        Toast.makeText(fragment.activity, "Message Loaded", Toast.LENGTH_LONG).show()
+    }
+})
+```
+
+
+
+##### `MessageSentListener`
+`MessageSentListener` is an interface for callbacks to be involved when a message is sent.
+
+- `onBeforeMessageSent` is called before a message is sent to Skygear Chat server.
+- `onMessageSentSuccess` is called after a message is sent.
+- `onMessageSentFailed` is called after a message is failed.
+
+
+##### `MessageFetchListener`
+`MessageFetchListener` is an interface for callbacks to be involved when messages of a conversation are fetched.
+
+- `onBeforeMessageFetch ` is called before messages are fetched.
+- `onMessageFetchSuccess` is called after messages are fetched.
+- `onMessageFetchFailed ` is called after fetching is failed.
+
+##### `ConnectionListener`
+`ConnectionListener` is an interface for callbacks to be involved when connection to server is changed.
+
+- `onClose` is called when pubsub connection is closed.
+- `onOpen` is called when pubsub connection is established.
+
+##### Intent Keys for each type of listeners
+
+|Key|Listener|
+|---|--------|
+|`MessageSentListenerIntentKey`|MessageSentListener|
+|`MessageFetchListenerIntentKey`|MessageFetchListener|
+|`ConnectionListenerIntentKey`|ConnectionListener|
+
+
+#### Title Option
```

Comment:
Not sure if I got it correctly, but if title option has nothing to do with event hooks, could we move it under the "Attribute Customisation" section? So that I don't have to jump between sections to understand how to customise the title of the conversation while I want to learn about how to customise the avatar for example.